clang/test/PCH/race-condition.cpp

```diff
@@ -0,0 +1,41 @@
+// RUN: %clang_cc1 -fallow-pch-with-compiler-errors -std=c++20 -x c++-header -emit-pch %s -o %t -verify
+// RUN: %clang_cc1 -fallow-pch-with-compiler-errors -std=c++20 -include-pch %t %s -verify
```

Comment:
In the `main` branch it didn't fail but simply hangs. Better than nothing though.